python/gym_jiminy/toolbox/gym_jiminy/toolbox/math/generic.py

```diff
@@ -59,18 +59,128 @@ def quat_to_rpy(quat: np.ndarray) -> np.ndarray:
     """Compute the Yaw-Pitch-Roll Euler angles representation of a single or a
     batch of quaternions.
 
-    :param quat: 1D array or a 2D array whose first dimension corresponds to
-                 the number of individual quaternions, and the second to the 4
-                 coordinates [qx, qy, qz, qw].
+    :param quat: 1D array, a 2D array or nD array whose first dimensions
+                 corresponds to the number of individual quaternions, and the
+                 last to the 4 coordinates [qx, qy, qz, qw].
     """
-    qx, qy, qz, qw = np.atleast_2d(quat)[:, -4:].T
+    qx, qy, qz, qw = np.atleast_2d(quat)[..., -4:].T
     roll = np.arctan2(2 * (qw * qx + qy * qz), 1.0 - 2 * (qx * qx + qy * qy))
     pitch = -np.pi / 2 + 2 * np.arctan2(
         np.sqrt(1 + 2 * (qw * qy - qx * qz)),
         np.sqrt(1 - 2 * (qw * qy - qx * qz)),
     )
     yaw = np.arctan2(2 * (qw * qz + qx * qy), 1.0 - 2 * (qy * qy + qz * qz))
-    rpy = np.stack((roll, pitch, yaw), axis=-1)
+    rpy = np.stack((roll.T, pitch.T, yaw.T), axis=-1)
     if quat.ndim == 1:
         return rpy[0]
     return rpy
+
+
+@nb.jit(nopython=True, nogil=True, inline='always')
+def quat_to_matrix(quat: np.ndarray) -> np.ndarray:
+    """Compute the Rotation Matrix representation of a single or a
+    batch of quaternions.
+
+    :param quat: 1D array, a 2D array or nD array whose first dimensions
+                 corresponds to the number of individual quaternions, and the
+                 last to the 4 coordinates [qx, qy, qz, qw].
+    """
+    qx, qy, qz, qw = np.atleast_2d(quat)[..., -4:].T
```

Comment:
I guess there is something cleaner than just transpose everwhere. May be just reverse dims at the beginning and do the contrary at the end.